src/Xamarin.Android.Build.Tasks/Tasks/ProcessAssemblies.cs

```diff
@@ -46,82 +50,120 @@ public override bool RunTask ()
 				}
 			}
 
+			// We only need to "dedup" assemblies when there is more than one RID
+			if (RuntimeIdentifiers.Length > 1) {
+				Log.LogDebugMessage ("Deduplicating assemblies per RuntimeIdentifier");
+				DeduplicateAssemblies (output, symbols);
+			} else {
+				Log.LogDebugMessage ("Found a single RuntimeIdentifier");
+				SetMetadataForAssemblies (output, symbols);
+			}
+
+			OutputAssemblies = output.ToArray ();
+			ResolvedSymbols = symbols.Values.ToArray ();
+
+			// Set ShrunkAssemblies for _RemoveRegisterAttribute and <BuildApk/>
+			if (PublishTrimmed) {
+				var shrunkAssemblies = new List<ITaskItem> (OutputAssemblies.Length);
+				foreach (var assembly in OutputAssemblies) {
+					var dir = Path.GetDirectoryName (assembly.ItemSpec);
+					var file = Path.GetFileName (assembly.ItemSpec);
+					shrunkAssemblies.Add (new TaskItem (assembly) {
+						ItemSpec = Path.Combine (dir, "shrunk", file),
+					});
+				}
+				ShrunkAssemblies = shrunkAssemblies.ToArray ();
+			}
+
+			return !Log.HasLoggedErrors;
+		}
+
+		/// <summary>
+		/// Called when RuntimeIdentifiers has one item
```

Comment:
Aside: not sure we need XML docs for *private* methods in general, never mind a class that doesn't otherwise contain XML documentation at all…